api/agent/agent.go

```diff
@@ -737,100 +745,155 @@ func newSizerRespWriter(max uint64, rw http.ResponseWriter) http.ResponseWriter
 
 func (s *sizerRespWriter) Write(b []byte) (int, error) { return s.w.Write(b) }
 
-func (a *agent) runHot(ctx context.Context, call *call, tok ResourceToken, state ContainerState) {
+// nannyHotContainer is a helper function to monitor initialization channel, container context as well as
+// agent shutdown during the lifetime of the container.
+func (a *agent) nannyHotContainer(ctx context.Context, callerCtx context.Context, cancel context.CancelFunc, evictor *EvictToken, initialized chan struct{}, pulled chan struct{}) {
```

Comment:
we should be able to reduce the contexts here by having the caller of nanny handle the cancel, I don't see any usage other than the first defer. since we only need to listen on the done channel for caller, it may read easier to just pass the channel from `callerCtx.Done()` than the whole context (not 2 contexts in scope) - sorry, as you know this is a trigger for me...